pypika/queries.py

```diff
@@ -66,29 +69,51 @@ def __hash__(self):
         return hash(str(self.name))
 
 
-class Table(Selectable):
-    def __init__(self, name, schema=None, alias=None):
-        super(Table, self).__init__(alias)
-        self._table_name = name
-        self._schema = schema
+class Schema:
+    def __init__(self, name, parent=None):
+        self._name = name
+        self._parent = parent
+
+    def __eq__(self, other):
+        return isinstance(other, Schema) \
+               and self._name == other._name \
+               and self._parent == other._parent
 
     def get_sql(self, quote_char=None, **kwargs):
         # FIXME escape
+        schema_sql = format_quotes(self._name, quote_char)
 
-        parts = []
-        if self._schema:
-            schema = (self._schema,) \
-                if not isinstance(self._schema, (list, tuple)) \
-                else self._schema
+        if self._parent is not None:
+            return '{parent}.{schema}' \
+                .format(parent=self._parent.get_sql(quote_char=quote_char, **kwargs),
+                        schema=schema_sql)
 
-            parts += [s for s in schema]
+        return schema_sql
 
-        parts.append(self._table_name)
 
-        table_sql = '.'.join("{quote}{part}{quote}".format(part=part,
-                                                           quote=quote_char or '')
-                             for part in parts)
+class Table(Selectable):
+    def __init__(self, name, schema=None, alias=None):
+        super(Table, self).__init__(alias)
+        self._table_name = name
+
+        # This is a bit complicated in order to support backwards compatibility. It should probably be cleaned up for
+        # the next major release. Schema is accepted as a string, list/tuple, Schema instance, or None
+        self._schema = schema \
+            if isinstance(schema, Schema) \
```

Comment:
I normally prefer anything more than a single conditional expression to be split out into multiple if statements on different lines for readability

Reply:
pulled it out into it's own function